### PR TITLE
30 shelters with pending applications

### LINKED ISSUE #30: Shelters with Pending Applications
For this story, you should fully leverage ActiveRecord methods in your query.

As a visitor
When I visit the admin shelter index ('/admin/shelters')

┆Issue is synchronized with this [Trello card](https://trello.com/c/nAtpe4Eo) by [Unito](https://www.unito.io)
┆Attachments: <a href="https://github.com/aleish-m/adopt_dont_shop/issues/15">#15 Shelters with Pending Applications</a>

### PR DESCRIPTION
**Pull request by eriggs0207**

### CODE REVIEW

File: app/models/shelter.rb [hunk 1]
```diff
@@ -20,6 +20,13 @@ def self.order_by_desc_alphabetically
     find_by_sql("SELECT shelters.* FROM shelters ORDER BY name desc")
   end
 
+  def self.shelters_by_status
+    joins(pets:[:applicants]).where({ "applicants.status" => 'Pending'}).distinct
+    #     select shelters.name, pets.id, pets.name, applicants.first_name, applicants.id from shelters join pets on she
+    # lters.id = pets.shelter_id join applicant_pets on pets.id = applicant_pets.pet_id join applicants on applicants.id = applicant_pets.applica
+    # nt_id where applicants.status like 'Pending';
```
Comment (aleish-m) [inline app/models/shelter.rb:27]: This was really cool to figure out first in SQL and then figure out how that translates to AR with the multiple join tables. I'm really impressed with how we worked through this part.

File: db/seeds.rb [hunk 2]
```diff
@@ -7,14 +7,24 @@
...
+@applicant = Applicant.create!(first_name: 'John', last_name: 'Dough', street_address: '123 Fake Street', city: 'Denver',
+  state: 'CO', zip: 80205, description: "I'm awesome", status: 'pending')
+@applicant_1 = Applicant.create!(first_name: 'Jimmy', last_name: 'Dough', street_address: '567 Fake Street', city: 'Denver',
+  state: 'CO', zip: 80205, description: "I'm awesome", status: 'Pending')
+@applicant_2 = Applicant.create!(first_name: 'Johnny', last_name: 'Johnson', street_address: '17 Psuedo Street', city: 'Denver',
+  state: 'CO', zip: 80205, description: "I'm awesome", status: 'In Progress')
+@applicant_3= Applicant.create!(first_name: 'Timmy', last_name: 'Thompson', street_address: '57 Real Street', city: 'Denver',
+  state: 'CO', zip: 80207, description: "I'm not so awesome", status: 'Pending')
+@applicant_4= Applicant.create!(first_name: 'Tommy', last_name: 'Timson', street_address: '12227 Real Blvd', city: 'Colorado Springs',
+  state: 'CO', zip: 80909, description: "I'm the worst", status: 'Approved')
```
Comment (aleish-m) [inline db/seeds.rb:24]: If only we would have remembered to add these applicants right away in our model's test, that probably would have knocked 30 min of work time off of figuring out why we had an empty array. lol  😅

**Review by aleish-m — APPROVED** [review]
